renku/core/dataset/providers/api.py

```diff
@@ -215,3 +234,54 @@ def get_access_token_url(self) -> str:
     def export(self, **kwargs) -> str:
         """Execute export process."""
         raise NotImplementedError
+
+
+class ProviderCredentials(abc.ABC, UserDict):
+    """Credentials of a provider.
+
+    NOTE: A ``None`` value for a key means that it is not set; however, an empty value, "", means that the key was set.
```

Comment:
Together with the other comment on `None` instead of `""`. We do have `renku.core.util.util.NO_VALUE` (And `NoValueType`) for this purpose, which allows showing intent more clearly and also reflects this in the type signature.

Reply:
`""` is a valid value. So, I've changed `None` to `NO_VALUE`.